engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,41 +1514,46 @@ private static boolean serializeAllTable(
-            for (int ii = 0; ii < columnSources.length; ++ii) {
-                if (concurrentAttemptInconsistent()) {
-                    if (log.isDebugEnabled()) {
-                        final LogEntry logEntry = log.debug().append(System.identityHashCode(logIdentityObject))
-                                .append(" Bad snapshot before column ").append(ii);
-                        appendConcurrentAttemptClockInfo(logEntry);
-                        logEntry.endl();
-                    }
-                    return false;
-                }
-
-                final ColumnSource<?> columnSource = table.getColumnSource(columnSources[ii]);
+        final ExecutionContext executionContext = ExecutionContext.getContext();
+        final JobScheduler jobScheduler;
+        // TODO Check with Ryan if I should keep this check
+        // if (executionContext.getOperationInitializer() instanceof PoisonedOperationInitializer) {
+        // jobScheduler = new ImmediateJobScheduler();
+        // }
+        if (!executionContext.getOperationInitializer().canParallelize()) {
+            jobScheduler = new ImmediateJobScheduler();
+        } else {
+            jobScheduler = new OperationInitializerJobScheduler();
+        }
 
-                final BarrageMessage.AddColumnData acd = new BarrageMessage.AddColumnData();
-                snapshot.addColumnData[ii] = acd;
-                final boolean columnIsEmpty = columnsToSerialize != null && !columnsToSerialize.get(ii);
-                final RowSet rows = columnIsEmpty ? RowSetFactory.empty() : snapshot.rowsIncluded;
-                // Note: cannot use shared context across several calls of differing lengths and no sharing necessary
-                // when empty
-                final ColumnSource<?> sourceToUse = ReinterpretUtils.maybeConvertToPrimitive(columnSource);
-                acd.data = getSnapshotDataAsChunkList(sourceToUse, columnIsEmpty ? null : sharedContext, rows, usePrev);
-                acd.type = columnSource.getType();
-                acd.componentType = columnSource.getComponentType();
-                acd.chunkType = sourceToUse.getChunkType();
-
-                final BarrageMessage.ModColumnData mcd = new BarrageMessage.ModColumnData();
-                snapshot.modColumnData[ii] = mcd;
-                mcd.rowsModified = RowSetFactory.empty();
-                mcd.data = getSnapshotDataAsChunkList(sourceToUse, null, RowSetFactory.empty(), usePrev);
-                mcd.type = acd.type;
-                mcd.componentType = acd.componentType;
-                mcd.chunkType = sourceToUse.getChunkType();
+        final CompletableFuture<Boolean> waitForResult = new CompletableFuture<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Consumer<Exception> onError = err -> {
+            exception.set(err);
+            waitForResult.complete(Boolean.FALSE);
+        };
+        jobScheduler.iterateParallel(
```

Comment:
I think we only want to do this if:
1. Many rows are requested (1MM+)
2. At least one of the column sources is not an `InMemoryColumnSource`

Reply:
Re #1, a major goal is to accelerate cloud (S3) accesses for UI snapshots.  A test or flag that indicates we have high-latency access would also be a good test IMO.

Reply:
I don't think there is an easy way to verify if we a column source we are working with maps to a high latency source like S3, without making a more involved change, and even then, the underlying data might be cached in the S3 layer. So we cannot be sure. 
Right now, I have made the change for `InMemoryColumnSource` and 1MM+ rows. Will discuss with Ryan during review if we can do any other checks.